<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>665</width>
    <height>491</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>Resources/self-driving cute.jpg</normaloff>Resources/self-driving cute.jpg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item alignment="Qt::AlignTop">
       <widget class="QLabel" name="label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>665</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionResetDes"/>
    <addaction name="actionResetStart"/>
    <addaction name="actionAddObstacle"/>
    <addaction name="actionRemoveObs"/>
    <addaction name="separator"/>
    <addaction name="actionStart_Search"/>
   </widget>
   <widget class="QMenu" name="menuDisplay">
    <property name="title">
     <string>Display</string>
    </property>
    <addaction name="actionSearchOrder"/>
    <addaction name="actionPathLength"/>
    <addaction name="actionClearPath"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionShowAllPath"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuMethod">
    <property name="title">
     <string>Method</string>
    </property>
    <addaction name="actionAstar"/>
    <addaction name="actionDJ"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuDisplay"/>
   <addaction name="menuMethod"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="movable">
    <bool>true</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::NoToolBarArea</set>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionStart_Search"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionResetDes">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ResetDes</string>
   </property>
  </action>
  <action name="actionResetStart">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ResetStart</string>
   </property>
  </action>
  <action name="actionAddObstacle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>AddObs</string>
   </property>
  </action>
  <action name="actionRemoveObs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RemoveObs</string>
   </property>
  </action>
  <action name="actionSearchOrder">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>SearchOrder</string>
   </property>
  </action>
  <action name="actionPathLength">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>PathLength</string>
   </property>
  </action>
  <action name="actionClearPath">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ClearPath</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionScaling">
   <property name="text">
    <string>Scaling</string>
   </property>
  </action>
  <action name="actionAddStart">
   <property name="text">
    <string>AddObs</string>
   </property>
  </action>
  <action name="actionDisplaySearchTree">
   <property name="text">
    <string>DisplaySearchTree</string>
   </property>
  </action>
  <action name="actionDisplayPathLength">
   <property name="text">
    <string>DisplayPathLength</string>
   </property>
  </action>
  <action name="actionStart_Search">
   <property name="text">
    <string>Start Search</string>
   </property>
  </action>
  <action name="actionAstar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Astar</string>
   </property>
  </action>
  <action name="actionDJ">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Dj</string>
   </property>
  </action>
  <action name="actionShowAllPath">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ShowAllPath</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionStart_Search</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_btnStart_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowAllPath</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_chkShowAllPath_toggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionQuit_triggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionResetDes</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionResetDestination_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionResetStart</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionResetStart_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddObstacle</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionAddObs_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveObs</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionDelObs_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>95</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDisplayPathLength</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionShowSearchedPathTreeWithDistances_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDisplaySearchTree</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>on_actionShowSearchedPathTree_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_btnStart_clicked()</slot>
  <slot>on_chkShowAllPath_toggled(bool)</slot>
  <slot>on_actionQuit_triggered()</slot>
  <slot>on_actionSave_triggered()</slot>
  <slot>on_actionOpen_triggered()</slot>
  <slot>on_actionResetStart_triggered(bool)</slot>
  <slot>on_actionResetDestination_triggered(bool)</slot>
  <slot>on_actionAddObs_triggered(bool)</slot>
  <slot>on_actionDelObs_triggered(bool)</slot>
  <slot>on_actionShowSearchedPathTree_triggered(bool)</slot>
  <slot>on_actionNew_triggered()</slot>
  <slot>on_actionClearResult_triggered(bool)</slot>
  <slot>on_actionAbout_triggered()</slot>
  <slot>on_actionShowSearchedPathTreeWithDistances_triggered(bool)</slot>
  <slot>on_actionEnlargement_triggered()</slot>
  <slot>on_actionReduction_triggered()</slot>
  <slot>on_actionZoom100_triggered()</slot>
 </slots>
</ui>
